ToDoAppWithCoreData/Controllers/ListViewController.swift

```diff
@@ -0,0 +1,178 @@
+
+import UIKit
+import CoreData
+
+protocol ListViewControllerDelegate {
+    func tapSaveButton(item: String, detail: String, beginDate: Date, beginTime: Date, endDate: Date, endTime: Date, exitingItem: NSManagedObject?)
+}
+
+
+class ListViewController: UIViewController, UITextViewDelegate {
+    
+    public var item: String = ""
+    public var detail: String = ""
+    public var beginDate: Date!
+    public var beginTime: Date!
+    public var endDate: Date!
+    public var endTime: Date!
+    public var exitingItem: NSManagedObject?
+    
+    @IBOutlet var textFieldItem: UITextField!
+    @IBOutlet var detailTextView: UITextView!
+    @IBOutlet var saveButton: UIButton!
+    
+    let dateFormatter = DateFormatter()
+    let timeFormatter = DateFormatter()
+    
+    @IBOutlet var beginDatePicker: UIDatePicker!
+    @IBOutlet var beginTimePicker: UIDatePicker!
+    @IBOutlet var beginDateButton: UIButton!
+    @IBOutlet var beginTimeButton: UIButton!
+    @IBOutlet var endDatePicker: UIDatePicker!
+    @IBOutlet var endTimePicker: UIDatePicker!
+    @IBOutlet var endDateButton: UIButton!
+    @IBOutlet var endTimeButton: UIButton!
+    
+    var delegate: ListViewControllerDelegate?
```

Comment:
循環参照を防ぐためにweakつけましょう

```suggestion
    weak var delegate: ListViewControllerDelegate?
```